code/fs2netd/tcp_socket.cpp

```diff
@@ -237,9 +250,57 @@ int FS2NetD_SendData(char *buffer, int blen)
 {
 	int flags = 0;
 
+	// on Unix with MSG_NOSIGNAL, prevent SIGPIPE, which is generated if receiver has disconnected
+	// the MSG_NOSIGNAL flag doesn't work on OS X or Solaris
+	// on OS X, use setsockopt() with SO_NOSIGPIPE instead, see above
+#if defined(SCP_UNIX) && defined(MSG_NOSIGNAL)
+	flags |= MSG_NOSIGNAL;
+#endif // SCP_UNIX && MSG_NOSIGNAL
+
 	socklen_t to_len = sizeof(sockaddr);
 
-	return sendto(mySocket, buffer, blen, flags, (sockaddr*)&FS2NetD_addr, to_len);
+#if defined(SCP_UNIX) && !defined(SO_NOSIGPIPE) && !defined(MSG_NOSIGNAL) && _POSIX_REALTIME_SIGNALS > 0
+		// Unix platforms do not ignore SIGPIPE like Windows does.  This results in the standalone crashing eventually.
+		// Failing to use MSG_NOSIGNAL or SO_NOSIGPIPE, both of which are only on some platforms, we'll use this.
+		// Borrowed from:
+		// http://www.microhowto.info/howto/ignore_sigpipe_without_affecting_other_threads_in_a_process.html
+
+		// Step 1: Add SIGPIPE to the signal mask using pthread_sigmask.
+		sigset_t sigpipe_mask;
+		sigemptyset(&sigpipe_mask);
+		sigaddset(&sigpipe_mask, SIGPIPE);
+		sigset_t saved_mask;
+		if (pthread_sigmask(SIG_BLOCK, &sigpipe_mask, &saved_mask) == -1) {
+			ml_printf("Could not set mask with pthread_sigmask in FS2NetD_SendData");
+			exit(1);
+		}
+#endif // SCP_UNIX && !SO_NOSIGPIPE && !MSG_NOSIGNAL && _POSIX_REALTIME_SIGNALS > 0
+
+		// Step 2: Execute the library code which might raise SIGPIPE.
+		auto sent_chars = sendto(mySocket, buffer, blen, flags, (sockaddr*)&FS2NetD_addr, to_len);
+
+#if defined(SCP_UNIX) && !defined(SO_NOSIGPIPE) && !defined(MSG_NOSIGNAL) && _POSIX_REALTIME_SIGNALS > 0
+		// Step 3: Accept any pending SIGPIPE using sigtimedwait.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+		// Some older GCC versions seem to incorrectly warn on this initialization.
+		// The pragmas should be removable once GCC 5+ is in use everywhere.
```

Comment:
Well, it's 4.8.2, which has pretty good C++11 support I think so it's been working so far.  But as it lacks sanitizers and has bugs like this I'm hoping to see the platform move to 5+ soon.
https://gcc.gnu.org/gcc-4.8/cxx0x_status.html
Also, CentOS and openSUSE's latest stable versions are both 4.8.5 and the standalones have been running fine on those platforms so far.